io/net/ethtool_test.py

```diff
@@ -128,17 +133,77 @@ def test_ethtool(self):
+                default_set = default.copy()
+                elements = all([elem == '' for elem in value])
+                if elements:
+                    self.log.warn("Cannot set device channel for null")
                 else:
-                    self.fail("failed")
+                    for i in range(4):
+                        if default[i] != '':
+                            default[i] = ['0', '1', int(default[i])//2,
+                                          default[i], int(default[i])+1]
+                            for j in range(5):
+                                if value[i] != '':
+                                    cmd = "ethtool %s %s %s %s" % (
+                                            self.args, self.iface,
+                                            param[i], default[i][j])
+                                    result = process.run(cmd, shell=True,
+                                                         verbose=True,
+                                                         ignore_status=True)
+                                    if state is 'up':
+                                        ping = self.networkinterface.
+                                        ping_check(self.peer, count=5)
+                                        if ping is not None:
+                                            self.cancel("ping fail value %s \
+                                                    to %s parameter" % (
+                                                    default[i][j], param[i]))
+                                    err_msg1 = "no RX or TX channel"
+                                    err_msg2 = "count exceeds maximum"
+                                    if result.exit_status != 0:
+                                        if err_msg1 in result.stderr_text:
+                                            self.log.info("Cannot set %s \
+                                                    value on %s parameter" % (
+                                                    default[i][j], param[i]))
+                                        elif err_msg2 in result.stderr_text:
+                                            self.log.info("Cannot set %s \
+                                                    value on %s parameter" % (
+                                                    default[i][j], param[i]))
+                                        else:
+                                            self.fail("%s %s" % (
+                                                self.args, result.stderr_text))
+                    cmd = "ethtool %s %s %s %s %s %s %s %s %s %s" % (
+                        self.args, self.iface, param[0], value[0], param[1],
+                        value[1], param[2], value[2], param[3], value[3])
+                    ret = process.run(cmd, shell=True, verbose=True,
+                                      ignore_status=True)
+                    time.sleep(30)
+                    if ret.exit_status != 0:
```

Comment:
just a suggestion 

`if  process.system(cmd, shell=True, verbose=True,
                                      ignore_status=True) !=0`

Reply:
@PraveenPenguin will move this inside if block